picamera2/picamera2.py

```diff
@@ -105,6 +107,7 @@ def __init__(self, camera_num=0, verbose_console=None, tuning=None):
             self.preview_configuration = self.create_preview_configuration()
             self.still_configuration = self.create_still_configuration()
             self.video_configuration = self.create_video_configuration()
+            self.read_formats()
```

Comment:
I think I've changed my mind a little bit, based on what you said about all the extra log messages that come out. I think we should not call this ourselves but expose it as an extra function or property that users can access.

Maybe we could make it a property that gets filled when you first request it, and after that it's cached?

As a point of nomenclature, we tend to use "format" where we mean a combination of "Bayer order, bit depth and packing", so where we add other stuff (e.g. resolution, crop, frame rate info etc.) I'd like to call that something else, maybe "mode"? So perhaps the property becomes "sensor_modes" (that fits with sensor_format and sensor_resolution that we already have).